web: add `remember me` feature to IdentificationStage (#10397)
goauthentik/authentik: website/docs/flow/stages/identification/index.md

When enabled, any user identifier will be accepted as valid (as long as they match the correct format, i.e. when [User fields](#user-fields) is set to only allow Emails, then the identifier still needs to be an Email). The stage will succeed and the flow will continue to the next stage. Stages like the [Password stage](../password/index.md) and [Email stage](../email/index.mdx) are aware of this "pretend" user and will behave the same as if the user would exist.

## Enable "Remember me on this device" <span class="badge badge--version">authentik 2024.10+</span>

> When enabled, users will be given the option at login of having their username stored on the device. If selected, on future logins this stage will automatically fill in the username and fast-forward to the password field.  Users will still have the options of clicking "Not you?" and going back to provide a different username or disable this feature.

Reviewer: ```suggestion
When enabled, users are given the option at login to store their username on the device. If selected, on future logins this stage will automatically fill in the username and skip to the password field.  Users will still have the options of clicking "Not you?" and going back to provide a different username or disable this feature.
```

Reviewer: @Ken, is the username literally stored on that exact device? And by device, do we mean phone, table, laptop? I'm a little confised by that part.

Also, do we store the username anywhere else other than the device (i.e. our DB)?

Author: Yes, the username is literally stored on the specific device the user is using: phone, laptop, desktop, whatever.  It's in a long-term localstore, which associates the URL of the login page, the username, and a token to link to the two, so that we can pull it up.  It's a secure mechanism; it's the same one we use for controlling sessions.

And yes, we store the username in our database, that's how we look up their password for comparison.
